src/streaming.ts

```diff
@@ -89,38 +91,58 @@ export class StreamProxy extends Duplexify {
    * @param {Object} argument - the argument to be passed to the apiCall.
    */
   setStream(apiCall, argument) {
-    const stream = apiCall(argument, this._callback);
-    this.stream = stream;
-    ['metadata', 'status'].forEach(event => {
-      stream.on(event, (...args: Array<{}>) => {
-        args.unshift(event);
-        this.emit.apply(this, args);
+    const setStream = (stream) => {
```

Comment:
Where possible, could you add types here? Specifically calling out `stream`.

Reply:
Could you show an example of what you mean? Little green when it comes to TypeScript.

Reply:
Sure!  This line could be something like:

```ts
const setStream = (stream: ReadableStream) => {
...
}
```

Eventually, we will make the types required when #242 lands.  For now, it's honor system :)